tests/components/workday/test_init.py

```diff
@@ -0,0 +1,51 @@
+"""Test Workday component setup process."""
+from __future__ import annotations
+
+from homeassistant import config_entries
+from homeassistant.components.workday.const import (
+    DEFAULT_EXCLUDES,
+    DEFAULT_OFFSET,
+    DEFAULT_WORKDAYS,
+)
+from homeassistant.core import HomeAssistant
+
+from . import TEST_CONFIG_EXAMPLE_1, TEST_CONFIG_WITH_PROVINCE, init_integration
+
+
+async def test_load_unload_entry(hass: HomeAssistant) -> None:
+    """Test load and unload entry."""
+    entry = await init_integration(hass, TEST_CONFIG_EXAMPLE_1)
+
+    state = hass.states.get("binary_sensor.workday_sensor")
+    assert state
+
+    await hass.config_entries.async_remove(entry.entry_id)
+    await hass.async_block_till_done()
+
+    state = hass.states.get("binary_sensor.workday_sensor")
+    assert not state
+
+
+async def test_update_options(hass: HomeAssistant) -> None:
+    """Test we get the form in options."""
```

Comment:
docstring is wrong, and I think this test is weird.
Maybe it should be closer to:
```python
    entry = await init_integration(hass, TEST_CONFIG_WITH_PROVINCE)

    # TODO: assert a state with old settings

    # update the options (no need to use async_init/async_configure) it should trigger reload
    hass.config_entries.async_update_entry(config_entry, options=new_options)
    await hass.async_block_till_done()

    # TODO: assert a state with new settings
```